packages/fxa-auth-server/scripts/recorded-future/check-and-reset.ts

```diff
@@ -156,13 +161,52 @@ Limit: ${resultsPerPageLimit}
     }
   }
 
-  if (accountEmails.length === 0) {
+  if (accountLogins.length === 0) {
     console.log('No account emails found.');
     return 0;
   }
 
+  const credentialsLookupFn = createCredentialsLookupFn(client);
+  const cleartextCredentials = await credentialsLookupFn(accountLogins, {
+    first_downloaded_gte: firstDownloadedDateIsoString,
+  });
+  statsd.increment(
+    'recorded-future.identity.credentials-lookup.cleartext-total',
+    cleartextCredentials.length
+  );
+
+  const verifyPassword = createVerifyPasswordFn(
+    Password,
+    authDb.checkPassword.bind(authDb)
+  );
+
+  const defaultConcurrency = 100;
+  const queue = new PQueue({
+    concurrency: defaultConcurrency,
+    interval: 1000,
+    intervalCap: defaultConcurrency,
+  });
+
+  for (const foundCredentials of cleartextCredentials) {
+    await queue.onSizeLessThan(defaultConcurrency * 5);
+
+    queue.add(async () => {
+      const acct = accounts[foundCredentials.subject as string];
+      const passwordMatched = await verifyPassword(foundCredentials, acct);
+
+      if (passwordMatched) {
+        // @TODO maybe reset and send email
+        statsd.increment(
+          'recorded-future.identity.credentials-lookup.password-match'
+        );
+      }
+    });
+  }
+
+  await queue.onIdle();
```

Comment:
If you don't await queue add call on line 193, is there a small chance the queue could seem idle, because an add is in progress?